File: SortingAlgorithmsCpp/SortingAlgorithms/ShellSort.cpp
#include "ShellSort.hpp"

void shellSort(int arr[], int size) {
	for (int interval = size / 2; interval > 0; interval /= 2) {
		for (int i = interval; i < size; i++) {
			int key = arr[i];
			int j;
			for (j = i; j >= interval && arr[j - interval] > key; j -= interval) {
				arr[j] = arr[j - interval];
			}
			arr[j] = key;
		}
	}
}

File: SortingAlgorithmsPython/shellSort.py
def shell_sort(arr):
    interval = len(arr) // 2
    while interval > 0:
        for i in range(interval, len(arr)):
            key = arr[i]
            j = i
            while j >= interval and arr[j - interval] > key:
                arr[j] = arr[j - interval]
                j -= interval 
            arr[j] = key
        interval //= 2